Common/Data/Custom/Fred/Fred.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.IO;
+
+namespace QuantConnect.Data.Custom.Fred
+{
+    public class Fred : BaseData
+    {
+        /// <summary>
+        /// Return the URL string source of the file. This will be converted to a stream
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        /// String URL of source file.
+        /// </returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            if (isLiveMode)
+            {
+                return new SubscriptionDataSource($"http://cache.quantconnect.com/alternative/fred/{config.Symbol.Value.ToLowerInvariant()}.csv",
+                    SubscriptionTransportMedium.RemoteFile);
+            }
+
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "fred",
+                    $"{config.Symbol.Value.ToLowerInvariant()}.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile
+            );
+        }
+
+        /// <summary>
+        /// Parses the data from the line provided and loads it into LEAN
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">Line of data</param>
+        /// <param name="date">Date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>New instance of USEnergy</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            var csv = line.Split(',');
+
+            return new Fred
+            {
+                Symbol = config.Symbol,
+                Value = Parse.Decimal(csv[1]),
+                Time = Parse.DateTimeExact(csv[0], "yyyyMMdd")
```

Comment:
We need to be careful here with the `Time` property since the time that is reported may be during the day, and not at `Time` @ 00:00:00. Perhaps it releases at `Time` @ 17:00:00? 

Either way, depending on the time they report the data, we might have to add a delay of a whole day to the time if we can't pin down the time the data is released.

Reply:
There are no specifications about the time of the day each series is reported.

Reply:
for daily:
EndTime=Time.Add(TimeSpan.FromDays(1));